simulator-spring-boot/src/main/java/org/citrusframework/simulator/service/ScenarioExecutionQueryService.java

```diff
@@ -84,10 +85,51 @@ public ScenarioExecutionQueryService(EntityManager entityManager, ScenarioExecut
         this.scenarioExecutionRepository = scenarioExecutionRepository;
     }
 
+    @VisibleForTesting
     static boolean isValidFilterPattern(String filterPattern) {
         return HEADER_FILTER_PATTERN.matcher(filterPattern).matches();
     }
 
+    private static Specification<ScenarioExecution> withResultDetailsConfiguration(ResultDetailsConfiguration config) {
+        return (root, query, cb) -> {
+            assert query != null;
+
+            // Prevent duplicate joins in count queries
+            if (query.getResultType() == Long.class || query.getResultType() == long.class) {
+                return null;
+            }
+
+            root.fetch(ScenarioExecution_.testResult, JoinType.LEFT);
+
+            if (config.includeParameters()) {
+                root.fetch(ScenarioExecution_.scenarioParameters, JoinType.LEFT);
+            }
+
+            if (config.includeActions()) {
+                root.fetch(ScenarioExecution_.scenarioActions, JoinType.LEFT);
+            }
+
+            if (config.includeMessages() || config.includeMessageHeaders()) {
+                var messagesJoin = root.fetch(ScenarioExecution_.scenarioMessages, JoinType.LEFT);
+
+                if (config.includeMessageHeaders()) {
+                    messagesJoin.fetch(Message_.headers, JoinType.LEFT);
+                }
+            }
+
+            // Return no additional where clause
+            return null;
+        };
+    }
+
+    private static Specification<ScenarioExecution> withIds(List<Long> executionIds) {
+        return (root, query, builder) -> {
+            var in = builder.in(root.get(ScenarioExecution_.executionId));
```

Comment:
In expressions with an arbitrary number of elements are in general critical. Oracle for example only supports 1000 Elements. This may or may not be an issue here, depending on the use case. Might not be an issue because of paging. Just want to mention it here.

Reply:
note that this is a general problem on _all_ queries on _all_ resources. the "fetch where id in" is all over the place, because pagination and criteria queries don't go well together in spring.